components/brave_wallet/browser/eth_response_parser.h

```diff
@@ -43,6 +43,11 @@ bool ParseUnstoppableDomainsProxyReaderGetMany(
 bool ParseUnstoppableDomainsProxyReaderGet(const std::string& json,
                                            std::string* value);
 
+bool ParseERC721TokenUri(const std::string& json, GURL* url);
```

Comment:
just to make sure, is this parsing going through the JSON sanitizer?

Reply:
Right, all requests made to fetch data from the blockchain via RPC are made via the api_request_helper, which [automatically sanitizes](https://github.com/brave/brave-core/blob/1f18e8986b112672d1216c5a4c43ec20558c438d/components/api_request_helper/api_request_helper.cc#L138) the JSON responses.

Later, when we fetch the metadata JSON from HTTPS or IPFS URIs we get from the blockchain, we also use the api_request_helper, so the JSON returned will be automatically sanitized.

If the URI is a data URI no additional request is made, so in that case we manually sanitize that JSON - https://github.com/brave/brave-core/blob/d917df512f60cb0e1c1654f3c95ee3838d5aea66/components/brave_wallet/browser/json_rpc_service.cc#L1621-L1635.